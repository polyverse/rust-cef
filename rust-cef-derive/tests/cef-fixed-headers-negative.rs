#[macro_use]
extern crate rust_cef_derive;

#[cef_fixed_headers()]
struct MissingAttributeValues {}

#[cef_fixed_headers]
struct MissingAttributeList {}

#[cef_fixed_headers(Version = "0", SomethingElse = "1", Version = "2")]
struct InvalidHeader {}

#[cef_fixed_headers(Version)]
struct SingleIdent {}

#[cef_fixed_headers(Version + 10)]
struct Expr {}

#[cef_fixed_headers(Version, Name)]
struct MultipleIdents {}

#[cef_fixed_headers(Version = 0)]
struct NonStringHeader {}

#[cef_fixed_headers(Version = "0")]
union ApplyToUnion {}

#[cef_fixed_headers(Version = "0")]
fn ApplyToFn() -> () {}

#[cef_fixed_headers(Version = "0")]
const APPLY_TO_CONST: &str = "";

struct ApplyToField {
    #[cef_fixed_headers(Version = "0")]
    pub field: usize
}

#[cef_fixed_headers(Version = "0", Version = "2")]
struct DuplicateHeaders1{}

#[cef_fixed_headers(Version = "0")]
#[cef_fixed_headers(Version = "0")]
struct DuplicateHeaders2 {}

fn main() {

}