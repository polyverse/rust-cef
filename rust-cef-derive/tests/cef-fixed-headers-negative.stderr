error: 'cef_fixed_headers' specifies no fixed header values. Remove it or provide values. 'cef_fixed_headers' macro expects fixed headers provided in the following syntax: #[cef_fixed_headers(header1 = "value1", header2 = "value2", ...)]
 --> $DIR/cef-fixed-headers-negative.rs:4:1
  |
4 | #[cef_fixed_headers()]
  | ^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: 'cef_fixed_headers' specifies no fixed header values. Remove it or provide values. 'cef_fixed_headers' macro expects fixed headers provided in the following syntax: #[cef_fixed_headers(header1 = "value1", header2 = "value2", ...)]
 --> $DIR/cef-fixed-headers-negative.rs:7:1
  |
7 | #[cef_fixed_headers]
  | ^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: header name should be one of the following: Version,DeviceVendor,DeviceProduct,DeviceVersion,DeviceEventClassID,Name,Severity
  --> $DIR/cef-fixed-headers-negative.rs:10:36
   |
10 | #[cef_fixed_headers(Version = "0", SomethingElse = "1", Version = "2")]
   |                                    ^^^^^^^^^^^^^

error: 'cef_fixed_headers' macro expects fixed headers provided in the following syntax: #[cef_fixed_headers(header1 = "value1", header2 = "value2", ...)]
  --> $DIR/cef-fixed-headers-negative.rs:13:21
   |
13 | #[cef_fixed_headers(Version)]
   |                     ^^^^^^^

error: expected `,`
  --> $DIR/cef-fixed-headers-negative.rs:16:29
   |
16 | #[cef_fixed_headers(Version + 10)]
   |                             ^

error: 'cef_fixed_headers' macro expects fixed headers provided in the following syntax: #[cef_fixed_headers(header1 = "value1", header2 = "value2", ...)]
  --> $DIR/cef-fixed-headers-negative.rs:19:21
   |
19 | #[cef_fixed_headers(Version, Name)]
   |                     ^^^^^^^

error: 'cef_fixed_headers' fixed header values must be strings.
  --> $DIR/cef-fixed-headers-negative.rs:22:31
   |
22 | #[cef_fixed_headers(Version = 0)]
   |                               ^

error: 'cef_fixed_headers' attribute only applies to Structs or Enums.
  --> $DIR/cef-fixed-headers-negative.rs:25:1
   |
25 | #[cef_fixed_headers(Version = "0")]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: expected one of: `struct`, `enum`, `union`
  --> $DIR/cef-fixed-headers-negative.rs:29:1
   |
29 | fn ApplyToFn() -> () {}
   | ^^

error: expected one of: `struct`, `enum`, `union`
  --> $DIR/cef-fixed-headers-negative.rs:32:1
   |
32 | const APPLY_TO_CONST: &str = "";
   | ^^^^^

error: expected an inert attribute, found an attribute macro
  --> $DIR/cef-fixed-headers-negative.rs:35:5
   |
35 |     #[cef_fixed_headers(Version = "0")]
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0119]: conflicting implementations of trait `rust_cef::CefHeaderVersion` for type `DuplicateHeaders1`:
  --> $DIR/cef-fixed-headers-negative.rs:39:1
   |
39 | #[cef_fixed_headers(Version = "0", Version = "2")]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   | |
   | first implementation here
   | conflicting implementation for `DuplicateHeaders1`
   |
   = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0119]: conflicting implementations of trait `rust_cef::CefHeaderVersion` for type `DuplicateHeaders2`:
  --> $DIR/cef-fixed-headers-negative.rs:43:1
   |
42 | #[cef_fixed_headers(Version = "0")]
   | ----------------------------------- first implementation here
43 | #[cef_fixed_headers(Version = "0")]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `DuplicateHeaders2`
   |
   = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)
